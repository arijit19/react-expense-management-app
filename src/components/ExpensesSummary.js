import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import {Link} from 'react-router-dom';

import getExpenseTotal from '../Redux/selectors/ExpenseTotal.js'

export const ExpensesSummary = (props)=> {
    const word = props.expensesCount <= 1 ? 'expense' : 'expenses'
    return(
    <div className='page-header'>
        <div className='content-container'>
            <h1 className='page-header__title'>Viewing : <span>{props.expensesCount}</span>  {word}. Total Expense: <span>{numeral(props.expensesTotal).format('$0,0.00')}</span> </h1>
            <div className='page-header__actions'>
                <Link to='/create' className='button'>Add Expense</Link>
            </div>
        </div>
    </div>
)};

const mapStateToProps = (state)=> {    
    return {
        expensesTotal : getExpenseTotal(state.expenses),
        expensesCount : state.expenses ? state.expenses.length : 0
    }
}

export default connect(mapStateToProps)(ExpensesSummary);