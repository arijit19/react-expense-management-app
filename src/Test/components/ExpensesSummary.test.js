import React from 'react';
import {shallow} from 'enzyme'

import {ExpensesSummary} from '../../components/ExpensesSummary.js';
import expenses  from '../fixtures/Expense.js';
import getExpenseTotal from '../../Redux/selectors/ExpenseTotal.js'

test('Testing Expenses summary render component',()=>{
    const total = getExpenseTotal(expenses);
    const length = expenses.length;
    const wrapper = shallow(<ExpensesSummary expensesCount={length} expensesTotal={total}/>);
    expect(wrapper).toMatchSnapshot()
})