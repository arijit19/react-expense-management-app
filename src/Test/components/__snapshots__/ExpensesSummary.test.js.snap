// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Expenses summary render component 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing : 
      <span>
        3
      </span>
        
      expenses
      . Total Expense: 
      <span>
        $23,500.00
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
